src/ppl/include/ppl/IOPlacer.h

```diff
@@ -291,12 +305,21 @@ class IOPlacer
   std::set<int> ver_layers_;
   std::unique_ptr<TopLayerGrid> top_grid_;
 
+  std::unique_ptr<AbstractIOPlacerRenderer> ioplacer_renderer_;
+
   // simulated annealing variables
+  ppl::SimulatedAnnealing* annealing_;
+
   float init_temperature_ = 0;
   int max_iterations_ = 0;
   int perturb_per_iter_ = 0;
   float alpha_ = 0;
 
+  // simulated annealing debugger variables
+  bool annealing_debug_mode_ = false;
```

Comment:
Does this replicate the null/non-null status of annealing_ ?

Reply:
As annealing is going to remain local, annealing_debug_mode_ should be set to false after the annealing run?